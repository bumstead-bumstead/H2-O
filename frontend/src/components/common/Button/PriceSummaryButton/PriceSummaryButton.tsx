import type { ButtonHTMLAttributes } from 'react';
import styled from '@emotion/styled';

type Props = ButtonHTMLAttributes<HTMLButtonElement>;

function PriceSummaryButton({ children, ...restProps }: Props) {
  return <StyledButton {...restProps}>{children}</StyledButton>;
}

export default PriceSummaryButton;

const StyledButton = styled.button<Props>`
  ${({ theme }) => theme.typography.TextKRMedium14}
  color: ${({ theme }) => theme.colors.primary500};
  border: 1px solid ${({ theme }) => theme.colors.primary300};
  background-color: transparent;
  padding: 6px 16px;
  border-radius: 30px;

  &:hover {
    background-color: ${({ theme }) => theme.colors.skyBlueCardBg};
  }

  &:active {
    background-color: ${({ theme }) => theme.colors.primary500};
    color: white;
  }

  &:disabled {
    color: ${({ theme }) => theme.colors.gray600};
    background-color: transparent;
    border-color: ${({ theme }) => theme.colors.gray200};
    cursor: not-allowed;
  }
`;
