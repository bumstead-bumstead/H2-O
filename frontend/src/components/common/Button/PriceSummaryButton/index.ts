export { default as PriceSummaryButton } from './PriceSummaryButton';
