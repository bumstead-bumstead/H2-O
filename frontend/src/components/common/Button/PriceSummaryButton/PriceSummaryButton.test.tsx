import PriceSummaryButton from './PriceSummaryButton';
import { render, screen, userEvent, waitFor } from '@/tests/test-util';

describe('PriceSummaryButton 컴포넌트 테스트', () => {
  it('버튼이 정상적으로 렌더링된다.', () => {
    render(<PriceSummaryButton>버튼</PriceSummaryButton>);

    expect(screen.getByRole('button')).toBeInTheDocument();
  });

  it('버튼의 onClick 핸들러가 클릭시 정상적으로 호출된다.', async () => {
    const mockOnClick = vi.fn();
    render(<PriceSummaryButton onClick={mockOnClick}>버튼</PriceSummaryButton>);

    userEvent.click(screen.getByRole('button'));
    await waitFor(() => expect(mockOnClick).toBeCalledTimes(1));
  });

  it('버튼이 disabled인 경우 클릭할 수 없다.', async () => {
    const mockOnClick = vi.fn();
    render(
      <PriceSummaryButton disabled onClick={mockOnClick}>
        버튼
      </PriceSummaryButton>,
    );

    const button = screen.getByRole('button');
    expect(button).toBeDisabled();

    userEvent.click(button);
    await waitFor(() => expect(mockOnClick).not.toBeCalled());
  });

  describe('PriceSummaryButton 스냅샷 테스트', () => {
    it('버튼의 스냅샷이 일치한다.', () => {
      render(<PriceSummaryButton>버튼</PriceSummaryButton>);

      expect(screen.getByRole('button')).toMatchSnapshot();
    });
  });
});
