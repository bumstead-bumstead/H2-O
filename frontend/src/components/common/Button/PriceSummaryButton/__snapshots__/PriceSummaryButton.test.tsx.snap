// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PriceSummaryButton 컴포넌트 테스트 > PriceSummaryButton 스냅샷 테스트 > 버튼의 스냅샷이 일치한다. 1`] = `
.emotion-0 {
  font-family: Hyundai Sans Text KR,sans-serif;
  font-size: 14px;
  font-weight: 500;
  line-height: 22px;
  letter-spacing: -0.03em;
  color: #00397B;
  border: 1px solid #6688B0;
  background-color: transparent;
  padding: 6px 16px;
  border-radius: 30px;
}

.emotion-0:hover {
  background-color: #A2C7E74D;
}

.emotion-0:active {
  background-color: #00397B;
  color: white;
}

.emotion-0:disabled {
  color: #838588;
  background-color: transparent;
  border-color: #DADCDD;
  cursor: not-allowed;
}

<button
  class="emotion-0"
>
  버튼
</button>
`;
